portal/static/js/profile.js

```diff
@@ -0,0 +1,2784 @@
+                                            return item.code; });
+                                    }
+                                }
+                                if (!self.demo.data.timezone && item.url === self.modules.SYSTEM_IDENTIFIER_ENUM.timezone) {
+                                    self.demo.data.timezone = item.timezone ? item.timezone : "";
+                                }
+                            });
+                        }
+                        if (!self.demo.data.raceCodes) {
+                            self.demo.data.raceCodes = [];
+                        }
+                    }
+                    callback(data);
+                });
+            },
+            setUserSettings: function() {
+                if ($("#profileForm").length > 0) {
+                    this.subjectId = $("#profileUserId").val();
+                    this.currentUserId = $("#profileCurrentUserId").val();
+                    this.mode = "profile";
+                }
+                if ($("#aboutForm").length > 0) {
+                    this.subjectId = $("#iq_userId").val();
+                    this.currentUserId = $("#iq_userId").val();
+                    this.mode = "initialQueries";
+                }
+                var acoContainer = $("#accountCreationContentContainer");
+                if (acoContainer.length > 0) {
+                    this.currentUserId = $("#currentStaffUserId").val();
+                    this.mode = acoContainer.attr("data-account") === "patient" ? "createPatientAccount": "createUserAccount";
+                }
+            },
+            getOrgTool: function(callback) {
+                if (!this.orgTool) {
+                    var self = this;
+                    callback = callback || function() {};
+                    this.orgTool = new (this.modules.orgTool) ();
+                    this.orgTool.init(function(data) {
+                        if (data && !data.error) {
+                            self.orgsData = data;
+                        }
+                        callback(data);
+                    });
+                    this.orgsList = this.orgTool.getOrgsList();
+                }
+                return this.orgTool;
+            },
+            isConsentEditable: function() {
+                var isStaff = this.currentUserRoles.indexOf("staff") !== -1;
+                var isPatient = this.currentUserRoles.indexOf("patient") !== -1;
+                var isEditableByStaff = this.settings.hasOwnProperty("CONSENT_EDIT_PERMISSIBLE_ROLES") && this.settings.CONSENT_EDIT_PERMISSIBLE_ROLES.indexOf("staff") !== -1;
+                var isEditableByPatient = this.settings.hasOwnProperty("CONSENT_EDIT_PERMISSIBLE_ROLES") && this.settings.CONSENT_EDIT_PERMISSIBLE_ROLES.indexOf("patient") !== -1;
+                return this.isAdmin() || (isStaff && isEditableByStaff) || (isPatient && isEditableByPatient);
+            },
+            isTestPatient: function() {
```

Comment:
this might be better checked directly, could be misconstrued with `test` role...